planner/core/rule_partition_processor.go

```diff
@@ -444,7 +464,33 @@ func (l *listPartitionPruner) findUsedListPartitions(conds []expression.Expressi
 			}
 			used[partitionIdx] = struct{}{}
 		} else {
-			return l.fullRange, nil
+			if len(r.HighVal) != len(exprCols) || r.IsFullRange() {
+				return l.fullRange, nil
+			}
+			if r.LowVal[0].Kind() == types.KindMinNotNull {
+				r.LowVal[0] = types.GetMinValue(exprCols[0].GetType())
```

Comment:
```suggestion
				r.LowVal[0] = types.GetMinValue(pruneExpr.GetType())
```

Should use the expression `RetType` instead of the column type.

Reply:
Oh no, ignore this, my fault.